src/value/native_writer.rs

```diff
@@ -0,0 +1,114 @@
+use crate::types::integer::IntAccess;
+use crate::value::writer::ElementWriter;
+use crate::value::{Element, Sequence, Struct, SymbolToken};
+use crate::{IonResult, IonType, RawSymbolTokenRef, Writer};
+
+/// Writes [Element] instances to the underlying [Writer] implementation.
+pub struct NativeElementWriter<W: Writer> {
+    writer: W,
+}
+
+impl<W: Writer> ElementWriter for NativeElementWriter<W> {
+    type Output = W;
+
+    fn write<E: Element>(&mut self, element: &E) -> IonResult<()> {
+        self.write_element(None, element)
+    }
+
+    fn finish(mut self) -> IonResult<Self::Output> {
+        self.writer.flush()?;
+        Ok(self.writer)
```

Comment:
🗺️ Calling `finish()` on the `ElementWriter` consumes it and returns the wrapped `Writer`.